Inventory.md

```diff
@@ -0,0 +1,173 @@
+- [doc of Jörmungandr fee policy: format.md](https://github.com/input-output-hk/chain-libs/blob/master/chain-impl-mockchain/doc/format.md)
+- [cardano-ledger TxFeePolicy](https://github.com/input-output-hk/cardano-ledger/blob/0ad59ce46b3da8af6f6af97064e7c77cc7c2ac8b/cardano-ledger/src/Cardano/Chain/Common/TxFeePolicy.hs#L41-L59)
+
+
+### Protocol Parameter Updates
+
+Jörmungandr does not fully support protocol parameter updates. If protocol parameters can change, the wallet needs to be aware.
+
+- Jörmungandr: static (updates not implemented)
+- Haskell Byron: static (updates possible, but not likely)
+- Haskell Shelley: dynamic (updates possible, wallet should handle this)
+
+### Blocks
+
+The Jörmungandr wallet uses the `BlockHeader = (slotId, blockHeight, hash, prevHash)` record to identify points on the chain.
+
+The Byron Haskell node only *needs* `Point ByronBlock = Origin | At (slotNo, hash)` to start giving us blocks. The wallet can retrieve `blockHeight` and `prevHash` from block headers. But those two fields are not needed to initiate the chain-sync protocol. Small ❌. 
+
+We might need to convert the genesis point `Origin`, to a `BlockHeader` if we are to persist it in the DB ❌
+
+Byron `SlotNo` counts slots from genesis in a single number. The current Jörmungandr-compatible wallet `SlotId` counts the epoch AND the slot in that epoch. ❌
+
+In Jörmungandr a slot can only be inhabited by a single block. On the Haskell side, epoch boundary blocks will share slot with the normal block that comes after it. ❌
+
+There is no limit on how far the Jörmungandr nodes can rollback. The Haskell Nodes have a limit: (`k` blocks). Haskell nodes have a clear separation between mutable and immutable parts of the chain.
+
+### Transactions
+
+Transaction inputs in Jörmungandr consist of (txHash, index, coinValue). In the Haskell Byron implementation they are only (txHash, index). ❌
+
+The addresses we get from the Haskell chain-sync protocol are not byte-strings, but rather in their deserialized form.
+
+### Delegation certificates (Shelley)
+
+There are delegation certificates in the Byron Haskell node.
+[haskell cardano-ledger `Certificate`](https://github.com/input-output-hk/cardano-ledger/blob/0ad59ce46b3da8af6f6af97064e7c77cc7c2ac8b/cardano-ledger/src/Cardano/Chain/Delegation/Certificate.hs#L87-L99<Paste>)
+I think they might be there for compatibility and be nothing we should concern ourselves with.
+
+The Jörmungandr wallet looks at delegation certificates to determine if, and to what it is delegating. This is pointless in Byron.
+
+### Pool registrations (Shelley)
+
+The wallet looks at pool registration certificates to collect metadata (pool-owners), for when it needs to list stake pools.
+
+### Block producer (Shelley)
+
+The stake pool metrics worker needs to know the minter of each block. In Byron there are no pools, so the worker would be pointless.
```

Comment:
Yes, and we've built this whole stake pool separately for a reason :) 
This way, we can rather easily create a server / API which does not implement anything regarding pools (modulo some tweaking in the way the "server" gets started).

Reply:
Yes 👍 

With a related potential problem, though:

> If `follow` is removed from core, `monitorStakePools` breaks (or will have be made jormungandr-specific).

Will try to elaborate elsewhere.